bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoHandler.java

```diff
@@ -296,135 +329,157 @@ public void onValueReceived(@Nullable String variable, @Nullable String value, @
-                service.removeSubscription(this, subscription);
-            }
+        synchronized (upnpLock) {
+            UpnpIOService localservice = service;
+            if (localservice != null) {
+                if (localservice.isRegistered(this)) {
+                    logger.debug("Removing WeMo GENA subscription for '{}'", getThing().getUID());
+                    ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+                    String subscription = BASICEVENT;
+
+                    if (subscriptionState.get(subscription) != null) {
+                        logger.debug("WeMo {}: Unsubscribing from service {}...", getUDN(), subscription);
+                        localservice.removeSubscription(this, subscription);
+                    }
 
-            if (thingTypeUID.equals(THING_TYPE_INSIGHT)) {
-                subscription = "insight1";
-                if (subscriptionState.get(subscription) != null) {
-                    logger.debug("WeMo {}: Unsubscribing from service {}...", getUDN(), subscription);
-                    service.removeSubscription(this, subscription);
+                    if (THING_TYPE_INSIGHT.equals(thingTypeUID)) {
+                        subscription = INSIGHTEVENT;
+                        if (subscriptionState.get(subscription) != null) {
+                            logger.debug("WeMo {}: Unsubscribing from service {}...", getUDN(), subscription);
+                            localservice.removeSubscription(this, subscription);
+                        }
+                    }
+                    subscriptionState = new HashMap<>();
+                    localservice.unregisterParticipant(this);
                 }
             }
-            subscriptionState = new HashMap<>();
-            service.unregisterParticipant(this);
-        }
-    }
-
-    private synchronized void onUpdate() {
-        ScheduledFuture<?> job = refreshJob;
-        if (job == null || job.isCancelled()) {
-            Configuration config = getThing().getConfiguration();
-            int refreshInterval = DEFAULT_REFRESH_INTERVALL_SECONDS;
-            Object refreshConfig = config.get("refresh");
-            if (refreshConfig != null) {
-                refreshInterval = ((BigDecimal) refreshConfig).intValue();
-            }
-            refreshJob = scheduler.scheduleWithFixedDelay(refreshRunnable, 0, refreshInterval, TimeUnit.SECONDS);
         }
     }
 
     private boolean isUpnpDeviceRegistered() {
-        return service.isRegistered(this);
+        UpnpIOService localservice = service;
+        if (localservice != null) {
+            return localservice.isRegistered(this);
+        }
+        return false;
     }
 
     @Override
     public String getUDN() {
         return (String) this.getThing().getConfiguration().get(UDN);
     }
 
+    public String getHost() {
+        UpnpIOService localservice = service;
```

Comment:
To be able to simply use this method for getting host without any local considerations, you could:
```suggestion
        String localHost = host;
        if (!localHost.isEmpty()) {
            return localHost;
        }
        UpnpIOService localservice = service;
```

Reply:
Changed